### PR TITLE
Fix the missing socket options for 32-bit.

### LINKED ISSUE #1917: Recent refactoring broke compiling on Android armv7, probably other 32-bit platforms too
### Expected behavior
Builds as normal.

### Actual behavior
[My daily CI for Android immediately notified me that this repo wasn't building for armv7](https://github.com/buttaface/swift-android-sdk/runs/3142060122?check_suite_focus=true#step:13:21), which I then tracked to #1907.

### Steps to reproduce

1. Build on or for a 32-bit platform: [my CI instructions can be seen here](
https://github.com/buttaface/swift-android-sdk/blob/main/.github/workflows/sdks.yml#L74).

### SwiftNIO version/commit hash

latest HEAD, f9b8f15

### Swift & OS version (output of `swift --version && uname -a`)

The official build of Swift 5.4.2 for centos8 running on Fedora Core 33, also Ubuntu 20.04 as seen on that daily CI.

I can build again with this simple patch copying over the missing constants for Android armv7 that are defined in `NIO/System.swift`, but I'll leave it up to you how to fix this:
```
diff --git a/Sources/NIOCore/BSDSocketAPI.swift b/Sources/NIOCore/BSDSocketAPI.swift
index 6f809484..787bec2e 100644
--- a/Sources/NIOCore/BSDSocketAPI.swift
+++ b/Sources/NIOCore/BSDSocketAPI.swift
@@ -56,6 +56,8 @@ import func WinSDK.inet_ntop
 import func WinSDK.inet_pton
 #elseif os(Linux) || os(Android)
 import Glibc
+let SO_RCVTIMEO = SO_RCVTIMEO_OLD
+let SO_TIMESTAMP = SO_TIMESTAMP_OLD
 
 private let sysInet_ntop: @convention(c) (CInt, UnsafeRawPointer?, UnsafeMutablePointer<CChar>?, socklen_t) -> UnsafePointer<CChar>? = inet_ntop
 private let sysInet_pton: @convention(c) (CInt, UnsafePointer<CChar>?, UnsafeMutableRawPointer?) -> CInt = inet_pton
```
My guess is this broke 32-bit linux too, as it needs the same constants.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

When pulling out the NIOBSDSocket later in #1907 we dropped some of the
socket option fixes for 32 bit platforms, which broke them.

Modifications:

Restore the socket options.

Result:

32-bit build should be working again.
Fixes #1917.

### CODE REVIEW

File: Sources/NIOCore/BSDSocketAPI.swift [hunk 1]
```diff
@@ -66,6 +66,15 @@ private let sysInet_ntop: @convention(c) (CInt, UnsafeRawPointer?, UnsafeMutable
 private let sysInet_pton: @convention(c) (CInt, UnsafePointer<CChar>?, UnsafeMutableRawPointer?) -> CInt = inet_pton
 #endif
 
+// Work around SO_TIMESTAMP/SO_RCVTIMEO being awkwardly defined in glibc.
+#if os(Android) && arch(arm)
+let SO_RCVTIMEO = SO_RCVTIMEO_OLD
+let SO_TIMESTAMP = SO_TIMESTAMP_OLD
+#elseif os(Linux)
+let SO_TIMESTAMP = CNIOLinux_SO_TIMESTAMP
```
Comment (glbrntt) [inline Sources/NIOCore/BSDSocketAPI.swift:74]: CI is very unhappy -- it looks like we're missing the CNIOLinux import?

**Review by glbrntt — CHANGES_REQUESTED** [review]

File: Sources/NIOCore/BSDSocketAPI.swift (hunk 1 above)
Comment (Lukasa) [reply Sources/NIOCore/BSDSocketAPI.swift:74]: Oh yup, seems likely.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Review by buttaface — APPROVED** [review]

**Review by glbrntt — APPROVED** [review]

**Pull request merged**